src/common/base/NebulaKeyUtils.cpp

```diff
@@ -27,9 +31,13 @@ std::string NebulaKeyUtils::edgeKey(PartitionID partId,
                                     EdgeRanking rank,
                                     VertexID dstId,
                                     EdgeVersion ev) {
+    static int32_t edgeMask = 0x40000000;
     std::string key;
     key.reserve(kEdgeLen);
-    key.append(reinterpret_cast<const char*>(&partId), sizeof(PartitionID))
+    type |= edgeMask;
+
+    int32_t item = (partId >> 8) | (NebulaKeyType::kData);
```

Comment:
This maybe be wrong. You discards the most significant byte when `>> 8`, and `bit or kData` at the least significant byte (which may overwrite the bits in partId). I think `int32_t item = (partId << 8) | (NebulaKeyType::kData);` is ok.

Reply:
It's reasonable. 

`int32_t` encode with little-encode but right shift and left shift maybe not dependence on encoded.

For example: `partID` is `0x0000000F` , `(partID << 8)` is `0x00000F00` .